src/basic/namespace-util.c

```diff
@@ -742,7 +742,7 @@ int is_idmapping_supported(const char *path) {
         if (r < 0)
                 return log_debug_errno(r, "Failed to acquire new user namespace for checking if '%s' supports ID-mapping: %m", path);
 
-        dir_fd = r = RET_NERRNO(open(path, O_RDONLY | O_CLOEXEC | O_NOFOLLOW));
+        dir_fd = r = RET_NERRNO(open(path, O_RDONLY | O_CLOEXEC));
```

Comment:
this is really not sufficient i fear.

setup_exec_directory() as the one relevant caller of is_idmapping_supported() does not follow symlinks for any of the other calls either. for example not for the chmod_and_chown() call a few lines above. (it does follow symlinks for the stat() call further down, as well as for path_chown_recursive() further down, admittedly.)

I think cleaning this up is always good (and it sounds like it would be ok to follo symlinks in this instance), but please do this comprehensively, and in particular, please add a small test case that validates this works, and ensures everything is chown()ed as intended in the end.